solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -1801,6 +1801,8 @@ private static Long readIntervalNs(String interval) {
   // in case of TLOG replica, if leaderVersion = zero, don't do commit
   // otherwise updates from current tlog won't copied over properly to the new tlog, leading to data
   // loss
+  // don't commit on leader version zero for PULL replicas as PULL should only get its index
+  // state from leader
   public static final String SKIP_COMMIT_ON_LEADER_VERSION_ZERO = "skipCommitOnLeaderVersionZero";
```

Comment:
I'm a bit dubious over how this parameter is used overall. When replicating from the leader, its value is always true. When recovering, it's true only for TLOG replicas (does it mean that the reason for setting it to true when replicating does not hold when recovering?).